services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/FlowCrudSpec.groovy

```diff
@@ -280,99 +280,37 @@ class FlowCrudSpec extends BaseSpecification {
-                }
-            ],
-            [
-                conflict: "no vlans both flows on same port on src",
-                makeFlowsConflicting: { FlowPayload dominantFlow, FlowPayload flowToConflict ->
-                    flowToConflict.source.portNumber = dominantFlow.source.portNumber
-                    flowToConflict.source.vlanId = 0
-                    dominantFlow.source.vlanId = 0
-                },
-                getError: { FlowPayload flowToError ->
-                    portError(flowToError.source.portNumber, flowToError.source.datapath, flowToError.id)
-                }
-            ],
-            [
-                conflict: "no vlans both flows on same port on dst",
-                makeFlowsConflicting: { FlowPayload dominantFlow, FlowPayload flowToConflict ->
-                    flowToConflict.destination.portNumber = dominantFlow.destination.portNumber
-                    flowToConflict.destination.vlanId = 0
-                    dominantFlow.destination.vlanId = 0
-                },
-                getError: { FlowPayload flowToError ->
-                    portError(flowToError.destination.portNumber, flowToError.destination.datapath, flowToError.id)
-                }
-            ],
-            [
-                conflict: "the same flowId",
-                makeFlowsConflicting: { FlowPayload dominantFlow, FlowPayload flowToConflict ->
-                    flowToConflict.id = dominantFlow.id
-                },
-                getError: { FlowPayload flowToError ->
-                    "Could not create flow: Flow $flowToError.id already exists"
-                }
-            ]
-        ]
+        where:
+        data << getConflictingData()
+    }
+
+    @Unroll("Unable to update flow with #data.conflict")
+    def "Unable to update flow with conflicting vlans or flow IDs"() {
+        given: "Two potential flows"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow1 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        def flow2 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+
+        def conflictingFlow2 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        data.makeFlowsConflicting(flow1, conflictingFlow2)
+
+        when: "Create two flows"
+        flowHelper.addFlow(flow1)
+        flowHelper.addFlow(flow2)
+
+        and: "Try updating the second flow which should conflict with the first one"
+        northbound.updateFlow(flow2.id, conflictingFlow2)
+
+        then: "Error is returned, stating a readable reason of conflict"
+        def error = thrown(HttpClientErrorException)
+        error.statusCode == HttpStatus.CONFLICT
+        error.responseBodyAsString.to(MessageError).errorMessage == data.getError(flow1, "update")
+
+        and: "Cleanup: delete flows"
+        [flow1, flow2].each { flowHelper.deleteFlow(it.id) }
+
+        where:
+        data << getConflictingData()[0..-2]
```

Comment:
Not critical, but a bit flaky solution, since someone may add additional cases to the end of getConflictingData without knowing that it is sequence-dependent (by default it is not common behavior for 'data provider'). It is better to have a base 'conflictingData' which is the same between tests and add a missing 'same flow id' case directly to the 'create' test.

Reply:
Done.